cli/src/main.rs

```diff
@@ -1,8 +1,89 @@
+use db::PgPool;
+use tokio;
+use clap::{Parser, Subcommand};
+use db::sqlx;
+use std::fs;
+use std::io::Write;
+use uuid::Uuid; // Ensure you have the `uuid` crate in your Cargo.toml
+
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+struct KnowledgeArgs {
+    #[command(subcommand)]
+    command: Command,
+}
+
+#[derive(Subcommand)]
+enum Command {
+    Login,
+    Signup,
+    DisplayUsers,
+}
+
 #[tokio::main]
 async fn main() -> color_eyre::Result<()> {
-    let _db_pool = db::setup_db_pool().await?;
+    let args = KnowledgeArgs::parse();
+    let db_pool = db::setup_db_pool().await?;
+
+    match args.command {
+        Command::Login => {
+            println!("Login not implemented yet")
+        },
+        Command::Signup => {
+            sign_up(&db_pool).await?;
+        },
+        Command::DisplayUsers => {
+            display_users(&db_pool).await?;
+        }
+    }
+
+    Ok(())
+}
+
+async fn persist_auth_session(user_id: Uuid) -> color_eyre::Result<()> {
+    if fs::metadata("auth.txt").is_ok() {
+        return Err(color_eyre::eyre::eyre!("auth.txt already exists"));
+    }
+
+    let mut file = fs::File::create("auth.txt")?;
```

Comment:
Nit: Since you already checked if the file exists it doesn't _really_ matter much, but since our intention is to always make a new file we might want to use `create_new` here instead
https://doc.rust-lang.org/std/fs/struct.File.html#method.create_new

This is cause `create` till truncate the file if one already exists, and `create_new` will error instead